shaping/shaper.go

```diff
@@ -14,6 +15,8 @@ import (
 // for each operation.
 type HarfbuzzShaper struct {
 	buf *harfbuzz.Buffer
+
+	fonts map[*font.Font]*harfbuzz.Font
```

Comment:
I think we need to make this an LRU. As it stands, there's no upper bound on the quantity of memory this will take if you use the same `HarfbuzzShaper` for a long time within a single application.

Reply:
What should be the capacity of the cache ?